Packs/JamfProtect/Integrations/JamfProtectEventCollector/JamfProtectEventCollector.py

```diff
@@ -525,32 +684,48 @@ def get_events_command(client, args) -> tuple:
      """
     limit = arg_to_number(args.get('limit')) or DEFAULT_LIMIT
     results = []
-    alert_events_with_time, audit_events_with_time = [], []
+    events_with_time: dict[str, list] = {
+        'Alert': [],
+        'Audit': [],
+        'Computer': []
+    }
     start_date, end_date = validate_start_and_end_dates(args)
 
-    alert_events, audit_events, _ = fetch_events(client=client, max_fetch_alerts=limit, max_fetch_audits=limit,
-                                                 start_date_arg=start_date,
-                                                 end_date_arg=end_date)
-
-    if alert_events:
-        alert_events = alert_events[:limit]
-        alert_events_with_time = add_time_field(alert_events)
-        results.append(
-            CommandResults(readable_output=tableToMarkdown("Jamf Protect Alert Events", alert_events_with_time),
-                           raw_response=alert_events_with_time))
-    if audit_events:
-        audit_events = audit_events[:limit]
-        audit_events_with_time = add_time_field(audit_events)
-        results.append(
-            CommandResults(readable_output=tableToMarkdown("Jamf Protect Audit Events", audit_events_with_time),
-                           raw_response=audit_events_with_time))
-    events = alert_events_with_time + audit_events_with_time
-    if events:
+    alert_events, audit_events, computer_events, _ = fetch_events(
```

Comment:
make `fetch_events` return an `EventResult`, and give `EventResult` a method called `as_dict(self)`, which would return `{'Alert': alert_events, ... : ... }`, and make iteration here cleaner (instead of creating a list of tuples)